napari/layers/shapes/_shape_list.py

```diff
@@ -1018,6 +1020,7 @@ def transform(self, index, transform):
         self.remove(index, renumber=False)
         self.add(shape, shape_index=index)
         self._update_z_order()
+        self.__dict__.pop('_bounding_boxes', None)
```

Comment:
I had to infer here that `cached_property` stores the cache in `self.__dict__`, and this line is clearing the cache. I think it would be good to abstract this as a one-line private method, `self._clear_bounding_box_cache()`?

And (optional) rather than manually calling this in each method that changes the data, should we not just connect it to an event that fires any time the data changes?

Reply:
It is not an evented model, and we already have performance problems with it.